api/src/main/java/org/openmrs/customdatatype/datatype/ProgramDatatype.java

```diff
@@ -9,23 +9,16 @@
  */
 package org.openmrs.customdatatype.datatype;
 
-import org.openmrs.OpenmrsMetadata;
-import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.OpenmrsObject;
 
 /**
- * This is a superclass for custom datatypes for OpenmrsMetadata
+ * This is a class for custom datatypes for programs
  * 
  * @since 1.12
  */
-public abstract class BaseMetadataDatatype<T extends OpenmrsMetadata> extends BaseOpenmrsDatatype<T> {
-	
-	/**
-	 * @see BaseOpenmrsDatatype#doGetTextSummary(Object)
-	 * @should use the name in summary instance
-	 */
+public class ProgramDatatype<T extends OpenmrsObject> extends BaseOpenmrsDatatype<T> {
```

Comment:
Thanks！ What would you suggest as an appropriate service? Since I really know little abt this part actually. Just in case you know that, or I can try to figure it out myself.